src/model/orca-pool.ts

```diff
@@ -0,0 +1,44 @@
+import { Connection, PublicKey } from '@solana/web3.js'
+import { OrcaPool } from '../public/types'
+import { OrcaPoolConfig } from '../public/pools'
+import { OrcaPoolParams } from './types'
+import { orcaPoolConfigs } from './pools'
+import { findAssociatedTokenAddress } from '../utils/find-associated-token-address'
+import { deserializeAccount } from '../utils/deserialize-account'
+import { decimalize } from '../utils/token-utils'
+
+export class OrcaPoolFactory {
+    getPool(connection: Connection, config: OrcaPoolConfig): OrcaPool {
+        return new OrcaPoolImpl(connection, orcaPoolConfigs[config])
+    }
+}
+
+class OrcaPoolImpl implements OrcaPool {
+    private connection: Connection;
+    private poolParams: OrcaPoolParams;
+
+    constructor(connection: Connection, config: OrcaPoolParams) {
+        this.connection = connection
+        this.poolParams = config
+    }
+
+    public async getLPBalance(user: PublicKey): Promise<number> {
+        const address = await findAssociatedTokenAddress(user, this.poolParams.poolTokenMint)
+
+        // TODO: SOL account handling
+
+        const accountInfo = await this.connection.getAccountInfo(address)
+
+        // User does not have a balance for this account
+        if (accountInfo == undefined) {
+            return 0
+        }
+
+        const result = deserializeAccount(accountInfo?.data)
+        if (result == undefined) {
+            throw new Error('Failed to parse user account for LP token.')
+        }
+
+        return decimalize(result.amount.toNumber(), this.poolParams.poolTokenDecimals)
```

Comment:
You want to be careful about converting `u64` directly into a number. Javascript numbers go up to 2^53, or 10^15. If a token has 9 decimals, `result.amount.toNumber()` will throw exceptions for amounts greater than 10^7. You should be able to use this helper function instead: https://github.com/orca-so/orca/blob/master/web/src/utils/format.ts#L95

Reply:
Another option is to use [decimal.js](https://github.com/MikeMcl/decimal.js/) before converting to a number. I kind of like that option better, since that library is fairly well documented and tested.

Reply:
Ah thanks. It's been a while since I worked with big numbers (reminds me of college). I've adjusted it to shift the decimals using Decimal.

Reply:
Nice. The conversion using Decimal looks clean!